shell/platform/android/io/flutter/embedding/engine/renderer/FlutterRenderer.java

```diff
@@ -1221,6 +1223,19 @@ public void setViewportMetrics(@NonNull ViewportMetrics viewportMetrics) {
       displayFeaturesType[i] = displayFeature.type.encodedValue;
       displayFeaturesState[i] = displayFeature.state.encodedValue;
     }
+    int cutoutOffset = viewportMetrics.displayFeatures.size() * 4;
+    for (int i = 0; i < viewportMetrics.displayCutouts.size(); i++) {
+      DisplayCutout displayCutout = viewportMetrics.displayCutouts.get(i);
+      displayFeaturesBounds[cutoutOffset + 4 * i] = displayCutout.bounds.left;
+      displayFeaturesBounds[cutoutOffset + 4 * i + 1] = displayCutout.bounds.top;
+      displayFeaturesBounds[cutoutOffset + 4 * i + 2] = displayCutout.bounds.right;
+      displayFeaturesBounds[cutoutOffset + 4 * i + 3] = displayCutout.bounds.bottom;
+      // Display cutouts always have type CUTOUT and state UNKNOWN.
+      displayFeaturesType[viewportMetrics.displayFeatures.size() + i] =
+          DisplayFeatureType.CUTOUT.encodedValue;
+      displayFeaturesState[viewportMetrics.displayFeatures.size() + i] =
```

Comment:
It seems weird to intentionally be setting unknown here. It seems to be what were were doing before but should we have added a state enum value for these?

Reply:
The current doc-comment for `UNKNOWN` states `The display feature is a cutout or this state is new and not yet known to Flutter.` The framework also currently expects cutouts to have a state of `unknown`.

Reply:
seems odd but out of scope for this pr.